frontends/web/src/components/transactions/transaction.tsx

```diff
@@ -151,35 +150,36 @@ const Status = ({
 
 type TAmountsProps = {
   amount: IAmount;
-  fee: IAmount;
+  deductedAmount: IAmount,
   type: TTransactionType;
 }
 
 const Amounts = ({
   amount,
-  fee,
+  deductedAmount,
   type,
 }: TAmountsProps) => {
   const txTypeClass = `txAmount-${type}`;
-  const sendToSelf = type === 'send_to_self';
   const sign = getTxSign(type);
+  const recv = type === 'receive';
+  const sendToSelf = type === 'send_to_self';
 
   return (
     <span className={`${styles.txAmountsColumn} ${styles[txTypeClass]}`}>
       {/* <data value={amount.amount}> */}
       <span className={styles.txAmount}>
         {sign}
         <Amount
-          amount={sendToSelf ? fee.amount : amount.amount}
-          unit={amount.unit}
+          amount={recv ? amount.amount : deductedAmount.amount}
+          unit={recv ? amount.unit : deductedAmount.unit}
         />
         <span className={styles.txUnit}>
           {' '}
-          {sendToSelf ? fee.unit : amount.unit}
+          {deductedAmount.unit}
         </span>
       </span>
       {/* </data> */}
-      <ConversionAmount amount={amount} type={type} />
+      <ConversionAmount amount={recv || sendToSelf ? amount : deductedAmount} type={type} />
```

Comment:
Same as with Amounts, it would probably make sense to pass amount, deductedAmount and type here and handle the type logic inside. Otherwise  we have the same confusion as before by sending in different data depending on the type, and inside the component doing different things based on the type.

`ConversionAmount` is also the wrong name for a component that in some cases does not show any conversions at all, but we already have the `Amount` component too, so it's not clear to me how to name these thing :sweat_smile: We can postpone that.

Reply:
> Same as with Amounts, it would probably make sense to pass amount, deductedAmount and type here and handle the type logic inside. Otherwise we have the same confusion as before by sending in different data depending on the type, and inside the component doing different things based on the type.
> 

Done

> `ConversionAmount` is also the wrong name for a component that in some cases does not show any conversions at all, but we already have the `Amount` component too, so it's not clear to me how to name these thing 😅 We can postpone that.

I can see that :) I didn't name it originally and I think if we want to rename it can be done in a different PR :)